### DISCUSSION TITLE
How to configure SQL Server for Elsa 3 ?

### DISCUSSION THREAD

**Discussion by lanfaboy** [elsa-workflows/elsa-core#5050] [Q&A]
#4972  Following this, prompts me that UseEntityFrameworkPersistence is not included

**Comment by glime-ai** [comment]
Based on the context provided, it seems like you are encountering an issue where the method `UseEntityFrameworkPersistence` is not recognized or included in your project. This could be due to several reasons, such as missing package references, incorrect namespaces, or version incompatibilities.

To resolve this issue, please ensure the following:

1. You have the correct version of the `Elsa.Persistence.EntityFrameworkCore` package installed that is compatible with your version of Elsa. Check the Elsa documentation or NuGet package repository for the correct version.

2. Make sure you are using the correct namespaces in your `Program.cs` or `Startup.cs` file. For example, you might need to include `using Elsa.Persistence.EntityFrameworkCore;` to access extension methods related to Entity Framework Core persistence.

3. If you are using a preview version of Elsa, as indicated in some of the context sources, ensure that the preview version you are using includes the `UseEntityFrameworkPersistence` method. Preview versions might have different APIs compared to stable releases.

4. Check the GitHub issues for Elsa (such as https://github.com/elsa-workflows/elsa-core/issues/4761) to see if others have encountered similar problems and if there are any suggested solutions or workarounds.

If you continue to face issues, consider reaching out to the Elsa community or support channels with specific details about your project setup, the versions of the packages you are using, and the exact error message you are receiving.

**Comment by glime-ai** [comment]
It seems like you are encountering an issue where the `UseEntityFrameworkPersistence` method is not recognized or included in your project. This could be due to a few reasons, such as missing package references or using an incorrect version of the Elsa package that does not contain the method.

To resolve this issue, please ensure that you have the correct version of the Elsa package that includes the `UseEntityFrameworkPersistence` method. You may need to install or update the `Elsa.Persistence.EntityFrameworkCore` NuGet package to the version that supports this method.

If you are using Elsa 2.x, make sure that your project references the `Elsa.Persistence.EntityFrameworkCore` package. If you are using Elsa 3.x (which is still in preview at the time of my knowledge cutoff in 2023), the persistence API might have changed, and you should refer to the latest documentation or GitHub issues for guidance on the new API.

You can check the Elsa documentation or the NuGet package page for the correct version to use. Additionally, you can look at the GitHub issues for any known issues or breaking changes that might affect the availability of the `UseEntityFrameworkPersistence` method.

If you continue to face issues, please provide more details about your project setup, including the version of Elsa you are using and the relevant parts of your `csproj` file, so that I can assist you further.
